ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -467,8 +465,16 @@ ledgerDbSwitch cfg numRollbacks trace newBlocks db =
             rollbackMaximum   = ledgerDbMaxRollback db
           , rollbackRequested = numRollbacks
           }
-      Just db' ->
-        Right <$> ledgerDbPushMany trace cfg newBlocks db'
+      Just db' -> case newBlocks of
+        [] -> pure $ Right db'
+        -- ^ no blocks to apply to ledger state, return current LedgerDB
```

Comment:
I'm not entirely sure, but I think the use of `-- ^` might lead to a haddock error.